### PR TITLE
[Docs][SLM] Add end-to-end tutorial for webLLM in SLM

### PR DESCRIPTION
**Pull request by CharlieFRuan**
Add end-to-end tutorial for webLLM using the SLM workflow. We go through two cases:
1. Only adding a model variant -- pointing to the convert weight page
2. Compiling a model library -- pointing to the compile model library page

### CODE REVIEW

**Comment by CharlieFRuan** [comment]
S4-F1 in https://github.com/mlc-ai/mlc-llm/issues/1420

cc @tqchen

File: docs/deploy/javascript.rst [hunk 1]
```diff
@@ -32,52 +32,163 @@ Note that the `WebLLM prebuilt webpage <https://webllm.mlc.ai/#chat-demo>`__ abo
...
+ - Quantization (e.g. ``q4f16_1``, ``q0f32``)
+ - Metadata (e.g. ``context_window_size``, ``sliding_window_size``, ``prefill-chunk-size``), which affects memory planning
+ - Platform (e.g. ``cuda``, ``webgpu``, ``iOS``)
+
+In cases where the model you want to run is not compatible with the provided MLC
+prebuilt model libraries (e.g. having a different quantization, a different
+metadata spec, or even a different model architecture), you need to build your
+own model library.
+
+Follow the page :ref:`compile-model-libraries`, specifically the ``WebGPU`` option. The page
```
Comment (tqchen) [inline docs/deploy/javascript.rst:163]: Would be great to copy the code over to here, so readers to not need to jump back and forth. Then add a reference to say if they would like to know more, they should look at the other page

File: docs/deploy/javascript.rst [hunk 2]
```diff
@@ -32,52 +32,163 @@ Note that the `WebLLM prebuilt webpage <https://webllm.mlc.ai/#chat-demo>`__ abo
...
+guides you through compiling ``RedPajama-INCITE-Instruct-3B-v1`` with ``q4f16_1``.
+After successfully following the page, you should end up with:
+
+.. code:: shell
+
+    ~/mlc-llm > ls dist/rp_q4f16_1
```
Comment (tqchen) [inline docs/deploy/javascript.rst:169]: use the naming convention `model_name-quant` (following the default naming scheme)

**Comment by tqchen** [comment]
Thanks @CharlieFRuan , added some comments that we can followup on

**Pull request merged**